org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/ScalaPreferences.scala

```diff
@@ -27,14 +34,26 @@ class ScalaPreferences extends FieldEditorPreferencePage with IWorkbenchPreferen
       }
     }
     addField(dataStoreField)
+
+    reconcilerField = new IntegerFieldEditor(ScalaPreferences.ReconcilerDelayId, "Reconciler delay (ms): ", getFieldEditorParent)
+    addField(reconcilerField)
   }
 
   override def initialize(): Unit = {
     super.initialize()
     dataStoreField.setStringValue(ScalaIdeDataStore.dataStoreLocation)
+    reconcilerField.setStringValue(IScalaPlugin().getPreferenceStore.getInt(ScalaPreferences.ReconcilerDelayId).toString)
   }
 
-  def init(workbench: IWorkbench): Unit = ()
+  override def init(workbench: IWorkbench): Unit = ()
+
+  override def performOk() = {
+    super.performOk()
+    val store = IScalaPlugin().getPreferenceStore
+    store.setValue(ScalaIdeDataStore.DataStoreId, dataStoreField.getStringValue)
+    store.setValue(ScalaPreferences.ReconcilerDelayId, reconcilerField.getStringValue)
+    true
```

Comment:
I just looked it up, `super.performOk()` returns always `true` too. No need to change then.